framework/src/fwk_module.c

```diff
@@ -449,6 +450,86 @@ int fwk_module_start(void)
     return FWK_SUCCESS;
 }
 
+static int fwk_module_stop_elements(struct fwk_module_context *module_ctx)
+{
+    int status;
+    const struct fwk_module *module;
+    unsigned int element_idx;
+
+    module = module_ctx->desc;
+
+    for (element_idx = 0; element_idx < module_ctx->element_count;
+         element_idx++) {
+
+        if (module->stop != NULL) {
```

Comment:
There is no need to do this check for each element. In fact, you only need to check this once for the module and don't call the function `fwk_module_stop_elements` if there is no `module->stop` defined.

Reply:
I have reused the same skeleton as fwk_module_start_module and fwk_module_start_elements which also do the check at both level. I wanted to stay consistent. Should I change both ?

Reply:
@vingu-linaro  it's up to you we can do the optimization now or later if you want.

Reply:
ok. So i'm going to add a patch to remove the useless check in start and will align the stop sequence

Reply:
After looking a bit more at the code, don't we have to loop on each element anyway in order to update its state ? This would mean that we have to call fwk_module_start/stop_elements for each elements even if module->stop/start == NULL in order to update the state.
So I'm going to let the code as it is for now because the change doesn't seem straightforward

Reply:
That's fine by me